### PR TITLE
Merge indexer docs

### PR DESCRIPTION
**Pull request by banool**
This PR rebasees the indexer-docs feature branch into main. Each commit here came from a separate PR. Each commit is independent, let's rebase this rather than merge.

### CODE REVIEW

**Review by rtso — APPROVED** [review]

File: developer-docs-site/sidebars.js [hunk 1]
```diff
@@ -359,6 +358,51 @@ const sidebars = {
         "tutorials/build-e2e-dapp/handle-tasks",
       ],
     },
+    {
+      type: "category",
+      label: "Learn about the Aptos Indexer",
```
Comment (davidiw) [inline developer-docs-site/sidebars.js:363]: for now, I'd probably still put this under advanced guides or integrate with aptos

agreed it probably belongs under infrastructure and nodes becomes more general...

**Review by davidiw — COMMENTED** [review]
before we land this, can we please setup redirects for recently moved content?

**Comment by banool** [comment]
Good call, that was on the todolist and subsequently forgotten 🏃‍♀️

File: developer-docs-site/sidebars.js (hunk 1 above)
Comment (banool) [reply developer-docs-site/sidebars.js:363]: Imo that's buried a little too deep. Anyway it doesn't really matter if we're gonna move it again in a few weeks.

**Comment by banool** [comment]
(I've gotta add another redirect and move the content again, this isn't ready to go)

**Comment by banool** [comment]
Okay we're good to go!

**Review by bowenyang007 — APPROVED** [review]

**Pull request merged**